src/reanimated2/layoutReanimation/animationBuilder/Keyframe.ts

```diff
@@ -0,0 +1,199 @@
+import { Easing, EasingFn } from '../../Easing';
+import { withDelay, withSequence, withTiming } from '../../animations';
+import {
+  AnimationFunction,
+  EntryExitAnimationBuild,
+  IEntryExitAnimationBuilder,
+  KeyFrameProps,
+  StyleProps,
+} from './commonTypes';
+
+export interface KeyFramePoint {
+  duration: number;
+  value: number | string;
+  easing?: EasingFn;
+}
+export interface ParsedKeyFramesDefinition {
+  initialValues: StyleProps;
+  keyframes: {
+    [key: string]: KeyFramePoint[];
+  };
+}
+
+export interface KeyFramesDefinition {
+  [key: number]: KeyFrameProps;
+}
+export class Keyframe implements IEntryExitAnimationBuilder {
+  durationV?: number;
+  delayV?: number;
+  definitions: KeyFramesDefinition;
+
+  /*
+    Keyframe definition should be passed in the constructor as the map
+    which keys are between range 0 - 100 (%) and correspond to the point in the animation progress.
+  */
+  constructor(definitions: KeyFramesDefinition) {
+    this.definitions = definitions;
+  }
+
+  private parseDefinitions(): ParsedKeyFramesDefinition {
+    /* 
+        for each style property contains an array with all its key points: 
+        value, duration of transition to that value, and optional easing function (defaults to Linear)
+    */
+    const parsedKeyFrames: {
+      [key: string]: KeyFramePoint[];
+    } = {};
+    const initialValues: StyleProps = this.definitions['0']
+      ? (this.definitions['0'] as StyleProps)
+      : {};
+    const duration: number = this.durationV ? this.durationV : 500;
+    const animationKeyPoints: Array<string> = Array.from(
+      Object.keys(this.definitions)
+    );
+    let previousKeyPoint = 0;
+
+    /* 
+       Because one of the assumptions is that each keyframe should contains
+       the same set of properties ( to make animation more consistent )
+       new entries in the parsedKeyFrames map can be added only in the first iteration
+       ( flag addIfNotExists is then set to true ).
```

Comment:
```suggestion
       Because one of the assumptions is that each keyframe should contain
       the same set of properties to make animation more consistent,
       new entries in the parsedKeyFrames map can be added only in the first iteration
       (flag addIfNotExists is then set to true).
```